src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,105 @@
 package core.basesyntax;
 
-/**
- * Feel free to remove this class and create your own.
- */
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
 public class HelloWorldTest {
 
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        // Clear storage before each test
+        core.basesyntax.db.Storage.people.clear();
+    }
+
+    @Test
+    public void register_validUser_userRegistered() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+    }
+
+    @Test
```

Comment:
The test case `register_existingLogin_userNotRegistered` should use a low-level method like `StorageDao.add()` to add `user1` to the storage instead of `registrationService.register(user1)`. This is to ensure that the setup does not depend on the correctness of the `register` method which is being tested.